packages/core/core-v2/src/Parcel.js

```diff
@@ -112,72 +108,74 @@ export default class Parcel {
     }
   }
 
-  async updateGraph() {
-    // Process all changes that may update the middle of the the graph
-    this.updateQueue.start();
-    await this.updateQueue.onIdle();
-    this.updateQueue.pause();
+  async updateGraph({signal}: BuildOpts) {
+    for (let [id, node] of this.graph.invalidNodes) {
+      this.processNode(node, {signal, shallow: true});
+    }
+
+    await this.queue.run();
   }
 
-  async completeGraph({ signal }: BuildOpts) {
-    // AssetGraph keeps track of nodes that represent files that have not been transformed
-    // and dependencies that have not been resolved so that rebuilds start from the leaves
-    // of the graph
+  async completeGraph({signal}: BuildOpts) {
     for (let [id, node] of this.graph.incompleteNodes) {
-      this.mainQueue.add(() => this.processNode(node, { signal }));
+      this.processNode(node, {signal});
     }
 
-    // Process and build out the edges of the graph
-    this.mainQueue.start();
-    await this.mainQueue.onIdle();
-    this.mainQueue.pause();
-
-    if (signal.aborted) throw abortError;
+    await this.queue.run();
   }
 
-  processNode(node: Node, { signal }: BuildOpts) {
+  processNode(node: Node, buildOpts: BuildOpts) {
     switch (node.type) {
-      case 'dependency': return this.resolve(node.value, { signal });
-      case 'file': return this.transform(node.value, { signal });
-      default: throw new Error('Invalid Graph');
+      case 'dependency':
+        return this.resolve(node.value, buildOpts);
+      case 'file':
+        return this.transform(node.value, buildOpts);
+      default:
+        throw new Error('Invalid Graph');
     }
   }
 
-  async resolve(dep: Dependency, { signal }: BuildOpts) {
-    // console.log('resolving dependency', dep);
-    let resolvedPath = await this.resolverRunner.resolve(dep);
+  resolve(dep: Dependency, {signal}: BuildOpts) {
+    return this.queue.add(async () => {
```

Comment:
I was a bit confused when reading this since the method changed from actually resolving to queuing a resolve. Can we either rename the method to something like `queueResolve` or keep `resolve` and do the queuing separately? Kinda like the latter, see next comment.

Reply:
I think I interpreted [this comment](https://github.com/parcel-bundler/parcel/pull/2000#discussion_r223875907) you made on the last PR as a suggestion to only add to the queue in one place, but yeah, the name of the function should probably change to imply we are queuing if that's the case. I kind of like just having `processNode`, `resolve`, and `tranform` just add to the queue directly too. I can change it back.